test: L2 fork tests

0xiamflux opened the pull request:
L2 Fork Testing Support
Adds capability to run L2 forked tests via justfile with minimal codebase changes. Tests are identified through Forge's path matching.
Required Environment Variables:

- L2_RPC_URL
- L2_BLOCK_NUMBER

Note: FORK_TEST takes precedence over L2_FORK_TEST when both are set.

mslipper committed:
op-deployer: Add doc describing bootstrap limitation (#14480)

* op-deployer: Add doc describing bootstrap limitation

* Update op-deployer/book/src/user-guide/known-limitations.md

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

---------

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

sebastianst committed:
go: Update op-geth to include latest SR 68ebaf81202ee7e21638232c54404074a49a63f6 (#14484)

protolambda committed:
op-signer, op-node: fix block signer utils (#14478)

* op-service: fix block signer utils

* op-service: improve block signer

axelKingsley committed:
increase the event executor queue limit 10x (#14483)

mslipper committed:
op-validator: Add op-validator (#14433)

* op-validator: Add op-validator

OP Validator is a CLI tool that returns a chain's level of compliance with the Standard Rollup Charter.

* redeploy, cr cleanups

* goimports

* docs

* docs

Inphi committed:
op-program: Add unichain mainnet to releases.json (#14485)

* op-program: Add unichain mainnet to releases.json

* remove governanceApproved for uni cannon64

Co-authored-by: mbaxter <meredith@oplabs.co>

* fix trailing comma

---------

Co-authored-by: mbaxter <meredith@oplabs.co>

protolambda committed:
op-service: RPC server wraps http-util server now, and factor out RPC handler (#14491)

* op-service: RPC server wraps http-util server now, and factor out RPC handler

* op-service: fix sub-test name

protolambda committed:
op-service: cleanup rw map utils (#14474)

* op-service: cleanup rw map utils

* op-service: fix doc comment

Co-authored-by: Axel Kingsley <axel.kingsley@gmail.com>

---------

Co-authored-by: Axel Kingsley <axel.kingsley@gmail.com>

maurelian committed:
feat: simplify DeployOPPU by removing unneeded inputs (#14379)

* feat: simplify by removing unneeded inputs

* Remove io contracts

* feat: get addrs from registry

* feat: get pdg blueprint from current opcm

* feat: add plan for getting fdg code

* feat: Add permissionlessDisputeGame bytecode and blueprint creation

* feat: Add blueprints checks

* fix: semgrep rule

* lint

* fix: use correct initcode

* fix: correct encoding of initCode

* semver

* fix imports

* feat: move IOPCM180 to /interfaces

* feat: print fdg initCodeHash

geoknee committed:
op-batcher: remove `ThrottleInterval` and split block loading and batch publishing into separate goroutines (#14244)

* op-batcher: overhaul throttling, reading and writing loops

* no longer evaluate throttling conditions on a ticker
* break main loop into "reading" and "writing"
* reading loop signals to throttling and writing when ALL blocks are loaded (this could be done in a more fine grained way, even when each block is loaded)
* these run concurrently

* improvements

* readloop unblocks writing loop once, and then writing loop goes forever (pausing when it runs out of data)

tests pass but timeout due to bad shutdown

* rename to prevent shadowing

* allow for receipt handling when receipt and err are both nil

* split shutdownCtx into producers and consumers

* throttling loop ranges over a channel, does not take a context

* processReceiptsLoop ranges over a channel, does not take a context

* unify wait groups

* unify contexts

* writeLoop uses a ticker instead of a sleep

* WIP: add throughput test for batcher

* make txQueue local, not global state

* reduce diff

* reduce diff further

* make pendingBytesUpdated a local, not global

* read and write loop communicate with a channel

no more ticker required

* rename

* rename

* abstract promptLoop

* rename

* update readme

* add diagram to readme

* remove test

I'm not sure it is adding any value. May return to it in future

* sendToThrottlingLoop uses mutex

* harmonize "*Loop returning" logs

* tidy

* remove TODO

* rip out ThrottleInterval config var

* add activeSequencerChanged channel and wire it up to onActiveSequencerChanged hook in active rollup provider

* set callback at runtime, not in constructor

* reinstate startup order

* tidy

* remove dead code

* remove unintentional change

* rename readLoop to blockLoadingLoop and writeLoop to publishingLoop

* improve diagram

* replace ThrottleInterval with ThrottleThreshold as enabling var

* remove onActiveProviderChanged arg from constructors

* protect callback invocation with nullity check

* lint

* Apply suggestions from code review

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

* remove ThrottelInterval from flags

* docs: mention active sequencer signal

* only attach callback if throttling is enabled

* increase buffer of activeSequencerChanged

means that we buffer a single event if the throttlingLoop is busy, even when using a try-send

* add buffer to pendingBytesUpdated

see previous commit

* reduce indentation

* remove dangling ref

* pass killCtx to publishingLoop and avoid accessing this context globally

* remove continue and always signal publishing loop

* remove unecessary mutex wrangle

* replace signalPublishingLoop method with trySignal fn

* extend TestRollupProvider_FailoverOnInactiveSequencer to cover callback fn

* use retryTimer in throttlingLoop

if RPC calls fail, they will be retried after retryInterval (or before, if another event triggers updateParams)

* op-batcher: improve active-seq-changed signalling setup

* remove unused state var

* rename blocksLoaded channel to publishSignal channel

---------

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

bitwiseguy committed:
devnet-sdk: add name to descriptors.DevnetEnvironment struct (#14492)

Hugoo committed:
docs: fix build instructions for op-node and op-proposer (#14340)

JosepBove committed:
Move to foundry v1 and benchmark (#14427)

* bump to foundry v1

* library harness failing

* rm return name

* fix internal revert tests

* rerun CI

* move to v1 tag

JosepBove committed:
Benchmark OPCM upgrade (#14360)

* benchmark upgrade call

* require to assert

* use lastCallGas

* Update packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

* change gas target to 20M

---------

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

mslipper committed:
ctb/op-deployer: Simplify superchain config/protocol versions validations (#14498)

* ctb/op-deployer: Simplify superchain config/protocol versions validations

While running the validators against newly-deployed chains, we discovered a bug where the superchain config's implementation cannot be checked unless the same proxy admin is shared between both the OP Chain and the superchain config. This only makes sense for OP Mainnet, so to simplify things I've removed the protocol version/superchain config checks from the standard validator. It will still check that the OP Chain is using the right superchain config, but it won't validate the config itself. We can use other tooling for that.

* checks

janjakubnanista committed:
feat(devnet-sdk): Add kurtosis data fetcher for externally deployed enclaves (#14446)

* wip: ktnative data fetcher

* wip: ktnative data fetcher

* fix: Only accept absolute paths

* chore: Add tests

* chore: inject dependencies instead of using globals

* chore: Comment fixes

* fix: Lint

blmalone committed:
fix: increase test-upgrade --rerun verbosity (#14502)

janjakubnanista committed:
chore: Add all predeployed addresses to devnet-sdk (#14507)

* chore: Add all predeployed addresses to devnet-sdk

* chore: Sort addresses

ajsutton committed:
op-dispute-mon: Update credit monitoring to handle new withdrawal path (#14496)

* op-dispute-mon: Load bond distribution mode.

* op-dispute-mon: Update credit monitoring to handle new withdrawal path.

* Simplify

geoknee committed:
op-batcher: add `TestBatchSubmitter_sendTx_FloorDataGas` and patch `driver.sendTx` (#14517)

* op-batcher: add TestBatchSubmitter_sendTx_FloorDataGas

* op-batcher: use floorDataGas for transactions if greater than intrinsicGas

* Update op-batcher/batcher/driver.go

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

* log error instead of ignoring

---------

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

smartcontracts committed:
feat: add IRI security review by OffbeatLabs (#14319)

Adds the Incident Response Improvements security review by offbeat
labs.

yashvardhan-kukreja committed:
feat: op-deployer cleanupAfterExit, unit tests cleanup and clean cache subcmd (#14488)

* feat: op-deployer clean cache command

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

* lint

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

* chore: cleaner dealing with test dir

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

* chore: refactor the IsolatedTestDirWithAutoCleanup and break import cycle

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

* chore: small cleanup

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

---------

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

sigma committed:
feat(op-program-svc): introduce new service to compute prestates (#14493)

* feat(op-program-svc): introduce new service to compute prestates

* test(op-program-svc): add tests

* Update kurtosis-devnet/op-program-svc/README.md

Co-authored-by: Inphi <mlaw2501@gmail.com>

---------

Co-authored-by: Inphi <mlaw2501@gmail.com>

bitwiseguy committed:
op-service: make signer client compatible with SetCodeTx type (#14489)

* op-service: make signer client compatible with SetCodeTx type

* use hexutil.U256 instead of hexutil.Big

* add explanatory comments

* Fix authorizationList json tag

Co-authored-by: protolambda <proto@protolambda.com>

* use helper fcn with panic for big.Int to uint256.Int

* use uint256.MustFromBig instead of custom helper

---------

Co-authored-by: protolambda <proto@protolambda.com>

smartcontracts committed:
maint: clean up security reviews table, update codeowners (#14523)

Cleans up the security reviews table because it had some errors
and a few missing entries. Updates codeowners so that the
security reviewers team can merge into the docs folder.

pauldowman committed:
chore: Update drippie config files with actual config from runbook (#14451)

ec2 committed:
Avoid lookup twice (#14354)

mbaxter committed:
vm-runner: Add metrics to track vm panics (#14504)

* Treat non-zero vm exits as a special case

* Only handle panics as a special case

* Add more tests

* Record panic metrics

* Tweak error message

NaijaCoderGirl committed:
update links in directory structure after PRs #12968 & #12828 (#14525)

smartcontracts committed:
feat: do not alias 7702 addresses (#14506)

Updates the OptimismPortal to not alias 7702 addresses. Updates
several other contracts to use a standard library for checking if
the msg sender is an EOA.

mslipper committed:
Revert "Added footer to op-deployer (#14374)" (#14528)

This reverts commit 9d94936939484b7898d29db581de5703f7685367.

geoknee committed:
op-batcher: always `updateCursorAndMetrics` when returning from `processBlocks()` (#14520)

* op-batcher: always updateCursorAndMetrics when returning from processBlocks()

* Update op-batcher/batcher/channel_manager.go

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

---------

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

janjakubnanista committed:
fix(devnet-sdk): Unique sentinels for devnet-sdk (#14514)

* fix: Unique markers for devnet-sdk

* chore: Add a test

maurelian committed:
feat: remove freeze from L1 and dispute contracts (#14438)

ajsutton committed:
op-challenger: Avoid loading duplicate leaves in same block (#14516)

* op-challenger: Avoid loading duplicate leaves.

* op-challenger: Sanity check length of fetched input data.

Provides a clearer error message if the fetch is incorrect rather than just InvalidProof() revert reason.

* op-challenger: Deduplicate blocks in contract bindings.

* Deduplicate block numbers in test setup now that the contract bindings handle it.

* Fix more tests.

smartcontracts committed:
feat: semgrep rules and check for reinitializer modifiers (#14534)

Adds semgrep rules and a new golang contracts check for
reinitializer modifiers. Important safety checks now that we are
using upgrade functions.

rebustron committed:
Update 404 link VERSIONING.md (#14519)

sebastianst committed:
all: update op-geth to v1.101500.1 (#14500)

* all: update op-geth to v1.101500.1

* op-batcher,op-e2e/system: Take floor data gas into account, fix AutoDA test

Fixes #14513

* Fix EIP-2935 implementation and tests

* op-program/client/l2: Fix chainContext.Config() to return actual cfg

* op-program/client/l2/test: Nilify chain config blob schedule

* update op-geth v1.101500.2-synctest.4

* Replace IntrinsicGas with FloorDataGas for calldata estimations

* Remove static MaxBlobsPerBlobTx constant

* update op-geth/v1.101500.2-rc.1

* revert usage of params.HistoryServeWindow

mbaxter committed:
Retry chain-id queries (#14539)

mslipper committed:
devnet-sdk: Fix casing in book theme (#14545)

mslipper committed:
chore: Fix another footer (#14548)

Inphi committed:
op-supervisor: Check message expiry (#14463)

* op-supervisor: Check message expiry

* add safe_update tests

* fix override expiry window description

* log warn when overriding expiry window

* lint

* fix tests

* Fix supervisor_checkMessages; use ErrConflict

* tweak supervisor_checkMessages API

* add supervisor_checkMessagesV2

* update op-geth dependency

* move override to depset config; fix checkMessagev1

* go.mod: drop diff again, op-geth was updated in develop branch

---------

Co-authored-by: protolambda <proto@protolambda.com>

Inphi committed:
op-challenger: Add Super DG contract caller (#14551)

* op-challenger: Add Super DG contract caller

* op-dispute-mon: Support Super FDG game types

Inphi committed:
op-node: Remove OverrideMessageExpiryTimeInterop from rollup.Config (#14554)

* op-node: Remove OverrideMessageExpiryTimeInterop from rollup.Config

* fix test fixture

maurelian committed:
feat: remove DeployOPCM (#14556)

pauldowman committed:
Add updated MT Cannon Spearbit audit (#14497)

geoknee committed:
op-batcher: remove `ChannelManager.CheckExpectedProgress()` and add channel timeout log (#14553)

* op-batcher: remove ChannelManager.CheckExpectedProgress

* op-batcher: add warning log when a channel times out on chain

zhiqiangxu committed:
optimize context usage (#14558)

mbaxter committed:
Add 64-bit tests for signExtend (#14543)

geoknee committed:
op-batcher: correctly track block metrics in `handleChannelInvalidated()` (#14561)

* op-batcher: correctly track block metrics in handleChannelInvalidated

Includes test which fails without the fix.

* op-batcher: log out channels which are dropped during handleChannelInvalidated()

* change to warn log for dropped channels

geoknee committed:
op-batcher: improve `computeSyncActions()` logging (#14563)

* improve computeSyncActions logging

* fixup test and make sure all cases run (!)

* use more friendly format for structured logger

ControlCplusControlV committed:
feat: Remove BlockNumber Dispute for SuperFaultDisputeGame + SuperPermissionedGame  (#14431)

* Initial copy

* Make tests work

* snapshots

* semantic version for pre-release

* my size was not size

* semver lock fixed oops

* added spec tests

* fix initialization test

* allow modifying super fdg

* Update packages/contracts-bedrock/src/dispute/SuperPermissionedDisputeGame.sol

Co-authored-by: Inphi <mlaw2501@gmail.com>

* introduced the deploy scripts

* added in new tests

* Update packages/contracts-bedrock/src/dispute/SuperPermissionedDisputeGame.sol

Co-authored-by: Inphi <mlaw2501@gmail.com>

* Update packages/contracts-bedrock/src/dispute/SuperFaultDisputeGame.sol

Co-authored-by: Inphi <mlaw2501@gmail.com>

* Patches

* fix tests

* remove unused errors

* new snapshots

* semver bump

* more snapshots

* vm()

* fixed spec tests

* snapshot + fix tests

* fix tests

* delete actors tests

* unused imports

* fix fork tests

* fix tests

* new test case

* use new prestate for deployment

* Remove Unused l2ChainId

* fix tests

* fix tests + snapshot

* missed one

* correct snapshot

---------

Co-authored-by: Inphi <mlaw2501@gmail.com>
Co-authored-by: Michael Amadi <amadimichaeld@gmail.com>

axelKingsley committed:
op-supervisor: independent reset preparation (#14444)

* interop: recovery draft

* Create consistencyState ; Wire required Backend Functions

* Full bisection ; block events during reset ; send reset signal

* unit test and fixes

* Cancel Reset if L1 Rewinds

* lint

* address comments ; fixes for tests

* Fix Unit Tests

* Add round-trip NodeID to Local Safe Updates (for ideal consistency tracking)

* op-supervisor,op-node: fixes to handle interop startup and resets

* refactor resetTracker ; fix E2E test

* Address PR Comments

---------

Co-authored-by: protolambda <proto@protolambda.com>

ajsutton committed:
op-program, op-challenger: Reduce number of steps per timestamp to 128 (#14572)

* op-program, op-challenger: Reduce number of steps per timestamp to 128

* Fix disputeed trace index.

protolambda committed:
op-test-sequencer: initial service scaffolding (#14270)

* op-test-sequencer: initial service framework

* op-test-sequencer: readme and justfile

sebastianst committed:
ci: only tag finalized Docker images as 'latest' (#14582)

alcueca committed:
Match OPCM version with monorepo release version (#14455)

* docs: We were still discussing some points about versioning

* fix: `isRC` is automatically set to false

* fix: reverted the versioning rules after the first change

* feat: explain why we have individual contract versioning

* feat: notes on deprecating `version()`

* docs: clearer branch denominations

* docs: when to remove `version()`

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

---------

Co-authored-by: alcueca <alberto@yield.is>
Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

tcrypt25519 committed:
interop: Recursive hazard detection (#14453)

* refactor,tweak: Add new HazardSet type and add recursive hazards.

* tweak: Update op-program consolidate.go to use new HazardSet.

* cleanup: Move map lookup into if.

* tests,cleanup: Use existing mock dependency set.

* fix

* tests,cleanup: Improve and cleanup HazardSet tests.

* op-program: Fix log index of executing messages in block

* fix: Re-add message expiry check from 1d05a2a.

* cleanup: Remove repeated log.

* tweak: Remove hazard recursion limit.

* docs: Update comment in (un)safe update tests to be accurate.

* tests: Update test comments and restore missing cycle tests.

* cleanup: Remove line break from fn sig.

* fix: Resolve merge issues in consolidate.go.

* fix: Use new message expiry window getter in consolidate.go.

* tests: Remove first skip in CascadeInvalidBlock.

* refactor: Cleanup HazardSet.build and split checks out into helpers.

* fix: Re-remove legacy expiry check after merge conflict.

---------

Co-authored-by: inphi <mlaw2501@gmail.com>

protolambda committed:
op-service: check if TLS is enabled and move middleware (#14571)

* op-service: check if TLS is enabled

* op-service: apply RPC middlewares globally

* op-service: apply RPC middlewares before RPC but after health

* op-service: test health before middleware

smartcontracts committed:
feat: bytecode verification script (#14589)

Adds a script that makes it easy to verify that the bytecode you
have in a local artifact matches the bytecode of a contract on a
live network.

ajsutton committed:
op-e2e: Write jwt.secret to a temp dir instead of into the source code tree. (#14575)

ControlCplusControlV committed:
default to the correct gameType (#14591)

tcrypt25519 committed:
Interop: Rewind tests (#14289)

* interop,fix: Treat missing L1 block as non-canonical.

* interop,tests: Add more Rewinder test cases.

* interop,tests: Remove unncessary error return.

geoknee committed:
op-batcher: introduce `PREFER_LOCAL_SAFE_L2` config var (#14587)

* op-batcher: introduce PREFER_LOCAL_SAFE_L2 config var

* lint

* Apply suggestions from code review

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

* lint

---------

Co-authored-by: Sebastian Stammler <seb@oplabs.co>

mslipper committed:
op-deployer: Start finalizing support for v2.0.0 (#14557)

* op-deployer: Start finalizing support for v2.0.0

* remove stdvertoml

* remove another usage of stdvertoml

* goimports

* update to latest SR

* bump SR again

* Add updated sepolia SV200 addresses

* update v200 validator

* fix test

* Update op-deployer/book/src/reference-guide/releases.md

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

---------

Co-authored-by: Maurelian <maurelian@protonmail.ch>
Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

ec2 committed:
mipsevm Memory Merklization Abstraction (#14292)

* Abstract the Merkle representation

* Implement asterisc's MPT

* migrate tests for mpt

* migrate tests for mpt

* copied benchmarks from asterisc

* fix failed merge

* Avoid pagelookup twice during setword invalidation

* fix state json codec test

* fix for singlethread too

* fix op-challenger test

* Remove MPT implementation

* address comments

* fix benchmark

leopardracer committed:
Update README.md (#14590)

zhiqiangxu committed:
remove devnetL1 files (#14619)

meyer9 committed:
feat: implement EIP-7251, EIP-7002 end-to-end test (#14253)

* feat: implement EIP-7251 end-to-end test

* Add missing import

* Fix rebase issue

* Fix lint

* Ensure requests hash is empty

* Move to action test

* Remove untested EIP from comment

meyer9 committed:
feat: add acceptance test for BLS precompiles (#13934)

* feat: add acceptance test for BLS precompiles

* Fix get proof block hash

* Revert get proof block hash change

* Test all BLS precompiles

* Add reference to test vectors

* Fix test name

* Add test before activation and remove extra geth test logic

protolambda committed:
op-node: drop stale todo in finality code (#14620)

protolambda committed:
op-test-sequencer: structure sub-responsibilities (#14501)

* op-test-sequencer: structure sub-responsibilities

* op-test-sequencer: update todo comment with issue number

mslipper committed:
ctb: Split up the OPCM (#14600)

* ctb: Split up the OPCM

* CR updates

* Update packages/contracts-bedrock/src/L1/OPContractsManager.sol

Co-authored-by: smartcontracts <kelvin@optimism.io>

* snapshots

---------

Co-authored-by: smartcontracts <kelvin@optimism.io>

mslipper committed:
ctb: Add upgrade 14 upgrader (#14612)

0xiamflux committed:
test: attempt to run forked tests from just file

ajsutton committed:
Remove redundant TODOs (#14610)

* op-e2e: Remove completed TODO

* op-test-sequencer: Remove TODO - ws support is not required here.

* op-supervisor: Remove TODO - RPC() method is intended to only return HTTP URLs.

* op-chain-ops: Remove TODO about making number of keys configurable.

The issue for this was closed with the decision that it didn't need to be configurable.

* op-program: Remove no longer relevant TODOs

* Undo script changes.

* Undo whitespace change.

Inphi committed:
op-e2e: cleanup unused MessageExpiryTime from interop recipe (#14634)

ajsutton committed:
op-dispute-mon: Support interop (#14598)

* op-dispute-mon: Add support for super root game types.

* op-dispute-mon: Switch to a single getter method for determining root type.

* op-dispute-mon: Don't init rollup client if rollup-rpc is not set.

Inphi committed:
contracts: Add solvency invariant test for Super FDG (#14555)

* contracts: Add solvency invariant test for Super FDG

* lint

ajsutton committed:
todo-checker: Fix handling of error codes. (#14611)

Now fails if it hits rate limits or other errors instead of silently ignoring.
Improves output to show all TODOs for closed issues.

ajsutton committed:
op-e2e: Update dispute game helpers to support SuperSystem and super cannon game types (#14574)

* op-e2e: Update dispute game helpers to support SuperSystem and super cannon game types.

* Switch back to permissioned cannon as default super system game type until OPCM supports super games
Don't deploy super games to pre-interop system

* Plural or genesis is actually geneses

* Add extra info in top half of super games.

* Move interop utils to new file.

* Add nil check.

maurelian committed:
Remove completed todo (#14640)

mslipper committed:
ops: Add docs around bailiff re-runs (#14632)

jsvisa committed:
contracts: rm not used format (#14593)

Signed-off-by: jsvisa <delweng@gmail.com>

Inphi committed:
op-program: Consolidate cascading block replacements (#14622)

* op-program: Consolidate cascading block replacements

Blocks containing invalid messages affect the cross-safety of its
dependencies. This is because once the block is replaced with a deposits-only block,
any messages it contained would no longer exist. And thus,
executing messages that referenced the non-existent messages become
invalid as well.

To deal with this, consolidation runs until all cross-depedencies are
reconciled against deposits-only updates.

* consolidate oracle

* op-e2e: intra-block cascade action case

* add unit tests for consolidate oracle

* always store state diff for the deposits-only block build

* add output root to block build db

* updated transition state view in consolidate oracle

* lint

sigma committed:
fix(devnet-sdk): allow in-place redeployment (#14412)

Currently, the last step of the deployment (storing the devnet
descriptor into the enclave) fails the 2nd time around.
As kurtosis API doesn't allow us to overwrite an artifact, we sidestep
the issue by storing successive generations of the descriptor.

Also adjust the environment fetcher to detect and load the last
generation.

0xOneTony commented on packages/contracts-bedrock/test/L2/forked/IsthmusUpgrade.t.sol:
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.25;
+pragma solidity 0.8.15;
```
Why do we change this?

0xiamflux replied:
It was just to match `Setup.sol` and other tests. Before that test file was not related to other tests so I didn't catch it being different.

0xOneTony commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -140,6 +140,11 @@ contract Setup {
         return vm.envOr("FORK_TEST", false);
     }
 
+    function isL2ForkTest() public view returns (bool) {
```
I think the name is confusing, having both `isForkTest` and `isL2ForkTest`. We basically do an upgrade of the L2 contracts so something like l2 upgrade fork test might be more suitable

0xiamflux replied:
Like it.

0xiamflux replied:
Addressed [here](https://github.com/defi-wonderland/optimism/pull/299/commits/f3060edf180264f4fa8127c624d9d48886b38298).

0xOneTony commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -156,6 +161,8 @@ contract Setup {
                 block.chainid == Chains.Sepolia || block.chainid == Chains.Mainnet,
                 "Setup: ETH_RPC_URL must be set to a production (Sepolia or Mainnet) RPC URL"
             );
+        } else if (isL2ForkTest()) {
```
This could be an `if` instead of `if-else if`

0xiamflux replied:
Yes, it can. Just wanted to highlight that it was supposed to be either L1 fork test or L2.

0xOneTony commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -140,6 +140,11 @@ contract Setup {
 
+    function isL2ForkTest() public view returns (bool) {
+        // Just in case we have both FORK_TEST and L2_FORK_TEST set
+        return !isForkTest() && vm.envOr("L2_FORK_TEST", false);
```
Why does it matter if we have both env vars true?

0xiamflux replied:
Currently L1 Fork tests skip the L2 setup.

pcw109550 committed:
devnet-sdk: Use geth logger instead of slog (#14644)

* Use testlog and geth log instead of slog

* Support devnet-sdk balance type for geth logging

* Update README for using testlog

agusduha commented:
What do you think on using the same ENV variables of the L1 fork and instead checking if it is a L2 fork with the block chain id?

zhiqiangxu committed:
add flags: `txmgr.max-basefee` and `txmgr.max-tip-cap` (#13382)

* add flags: txmgr.max-basefee and txmgr.max-tip-cap

* add TestMaxFees

* address comment

* mod error msg

mslipper committed:
ctb: Add contracts book (#14643)

maurelian committed:
Adress and remove todo (#14646)

0xiamflux commented:
> What do you think on using the same ENV variables of the L1 fork and instead checking if it is a L2 fork with the block chain id?

Like the idea.

dependabot[bot] committed:
dependabot(gomod): bump github.com/klauspost/compress (#14436)

Bumps [github.com/klauspost/compress](https://github.com/klauspost/compress) from 1.17.11 to 1.18.0.
- [Release notes](https://github.com/klauspost/compress/releases)
- [Changelog](https://github.com/klauspost/compress/blob/master/.goreleaser.yml)
- [Commits](https://github.com/klauspost/compress/compare/v1.17.11...v1.18.0)

---
updated-dependencies:
- dependency-name: github.com/klauspost/compress
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

smartcontracts committed:
feat: update message passing for EIP7623 (#14608)

* feat: update message passing for EIP7623

Updates the way that gas limits are handled in the message passing
system to account for EIP-7623. Most significant change is to
make the baseGas calculation inside of CrossDomainMessenger more
principled.

* fix: clean up XDM tests

* fix: apply EF byte change to all code

* fix: add OPPrestateUpdater to allow list

* fix: add changes to OPCM upgrade

0xiamflux committed:
test: update function name and condition for isL2UpgradeForkTest

0xiamflux commented:
> What do you think on using the same ENV variables of the L1 fork and instead checking if it is a L2 fork with the block chain id?

Addressed [here](https://github.com/defi-wonderland/optimism/pull/299/commits/f3060edf180264f4fa8127c624d9d48886b38298).

0xOneTony reviewed (commented):
I am wondering, is there a document in the repo that we should update regarding this new testing path in the setup? cc @agusduha

maurelian committed:
Add upgrade 13 audit (#14647)

protolambda committed:
op-supervisor: cleanup stale todo comment (#14648)

JosepBove committed:
Test upgrades on a list of chains (#14358)

* prepare upgrade parameters

* Archive rpcs

* change var type rerun

* rerun ci

* rerun ci

* bump block number

* remove context

* set cached block number

* try to solve error

* move block number to justfile

* update just file

* update blocknumber on upgrades ci

* remove last config edit

* fix config

* Update packages/contracts-bedrock/justfile

Co-authored-by: Maurelian <john@oplabs.co>

* move logic to pinnedblocknumber

* fix

* fix just file

* fix again

* back to just

* remove char to rerun CI

* readd char to rerun CI

* add env to pinned block

* back to og name for merge

* only op-mainnet

* use current block number

---------

Co-authored-by: Maurelian <john@oplabs.co>

yuwen01 committed:
isthmus: operator fee (#12166)

* feat: update IL1Block and info

* feat: update SystemConfig

* feat: add operator fee vault

* feat: handle transactions at Istmus activation

* feat: update GasPriceOracle

* feat: update deployment

* chore: update snapshots and semvers

* fix: GasPriceOracle nits

* fix: operator fee scalars ordering

* fix: handle feedbacks

* feat: remove operator fee vault constructor arguments

* chore: update e2e TestFees to handle operator fee

* chore: update snapshots

* fix: remove operator fee vault parameters on deployment

* feat: handle overflow for operator fee in GasPriceOracle

* chore: fee test

* chore: add e2e test to ensure operator fee doesn't ends up minting or burning eth

* fix: tests

* chore: use operator fee OP Geth branch for CI

* fix: tests

* fix(op-node): ordering of network upgrade txs

* fix: semver lock

* fix: sys config interop

* fix: semver lock

* fix: arithmetic return args

* fix: semver lock

fix: semver lock

fix: semver lock

fix: semver lock

fix: semver lock

* update system config interop

* fix: sys config interop semver lock

* fix: trailing newline

* chore: update tag

* fix: point to Isthmus by default

* fix: TestParseL1InfoDepositTxData

* Update test to catch Alice balance underflow

* fix: Test_Operator_Fee_Constistency

chore: update go.mod

* chore: update tag

* fix: operator fee are nil in receipts if both are 0

* chore: set the correct op-geth tag

* fix: handle sebastianst feedbacks

* fix: another round of feedbacks

* nit: useless cast

* chore: remove beta in contract versions

chore: update contracts versions

* chore: more comments on contract versions

* fix semgrep

* preinstall compilers

* schema

* indent

* add yes pipe

* ignore yes code

* try echo

* try wiping out the svm dir

---------

Co-authored-by: leruaa <aurelien.catinon@gmail.com>
Co-authored-by: refcell <abigger87@gmail.com>
Co-authored-by: Teddy Knox <teddyknox@gmail.com>
Co-authored-by: Matthew Slipper <me@matthewslipper.com>

leruaa committed:
feat: handle operator fee in OPCM (#14662)

* feat: handle operator fee in OPCM

* fix: upgrade version

* feat: check implementation address

zhiqiangxu committed:
op-deployer: `l1-rpc-url` is not needed when `deployment-target` is `genesis` (#13515)

* l1-rpc-url is only needed when deployment-strategy is live

* address comment

pcw109550 committed:
Decouple node from chain (#14660)

dpulitano committed:
op-conductor: Remove block time check for unsafe head progress (#14655)

* remove block time check for unsafe head progress

* fix test for this case

* slightly more succinct

* fix sequencer mock to stay healthy

meyer9 committed:
fix: remove requests processing step (#14623)

* fix: remove requests processing step

* Keep processing steps, but disable for Isthmus

* Update geth version

Inphi committed:
op-e2e: Configurable blocktime setup for interop (#14672)

geoknee committed:
all: update op-geth to v1.101503.0 (#14560)

* replace vm.AccountRef with common.Address

* op-chain-ops: replace prankRef by common.Address

* mise: update golangci-lint to 1.64.5

* op-chain-ops: fix prank handeCaller

* update to canonical op-geth version

maurelian committed:
ct docs: update release process without suffixes in contract semvers (#14682)

* ct docs: update release process without suffixes in contract semvers

* ct docs: discourage code freeze on trunk

* ct docs: update the additional release candidate process

* ct docs: precribe a method for avoiding semver collisions

mbaxter committed:
Fix broken link in versioning.md (#14688)

* Fix broken link

* Fix again :)

* Fix another broken link

zhiqiangxu committed:
op-node: continue sequencing when `L1TemporaryErrorEvent` happens (#14062)

* continue sequencing when L1TemporaryErrorEvent happens

* add TestSequencerL1TemporaryErrorEvent

* address comment

* fix test

ajsutton committed:
op-challenger: Add config option to set dependency set config (#14666)

* op-challenger: Add config option to set dependency set config

* Fix expected error message

ajsutton committed:
Add 1.5.0-rc.3 to releases.json (#14695)

ajsutton committed:
ci: Enforce correct formatting for TODOs. (#14639)

Fix the few TODOs that didn't correctly reference issues.
No requests to github are made when not checking closed issues. Ensures this runs fast and reliably in CI pre-merge.
Closed issues are then checked with a scheduled job.

ajsutton committed:
op-challenger: Update op-program executor to handle interop properly (#14667)

* op-challenger: Add config option to set dependency set config

* Fix expected error message

* op-challenger: Update op-program executor to handle interop inputs.

ajsutton committed:
Add ToSuper() method to convert super reponse to Super instance. (#14670)

* Add ToSuper() method to convert super reponse to Super instance.

Fix op-supervisor so it includes the Version in responses.

* op-proposer: Only enforce supported version of output roots on rollup source.

* op-challenger: Fix tests

0xOneTony committed:
Merge branch 'develop' of https://github.com/defi-wonderland/optimism into chore/sync

teddyknox committed:
Add devnet-sdk validators for fork configuration (#14668)

teddyknox committed:
Improve address management in kurtosis-devnet and devnet-sdk (#14703)

pcw109550 committed:
kt-devnet: add missing network params for supervisor (#14708)

zhiqiangxu committed:
show more error info when EngineAPIError is returned (#14673)

teddyknox committed:
Misc renames for clarity to fields in devnet-sdk and kurtosis-devnet (#14705)

* Improve address management in kurtosis-devnet and devnet-sdk

* Perform various renames for clarity in kurtosis-devnet and devnet-sdk

yashvardhan-kukreja committed:
fix(op-deployer): goreleaser linker flags for rendering version (#14344)

Signed-off-by: Yashvardhan Kukreja <yashvardhan@oplabs.co>

jsvisa committed:
feat: rm unnecessary generic type arguments (#12663)

* op-chain-ops: rm generic unnecessary type arguments

Signed-off-by: jsvisa <delweng@gmail.com>

* op-conductor: rm generic unnecessary type arguments

Signed-off-by: jsvisa <delweng@gmail.com>

* op-service: rm generic unnecessary type arguments

Signed-off-by: jsvisa <delweng@gmail.com>

* op-chain-ops: rm +1

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

protolambda committed:
ops: remove outdated legacy env rc example (#14724)

teddyknox committed:
Ingest L1 Chain ID into devnet-sdk (#14704)

* Improve address management in kurtosis-devnet and devnet-sdk

* Import L1 chain ID into devnet-sdk context

mbaxter committed:
Unfreeze MIPS contracts (#14726)

mbaxter committed:
cannon: Consolidate state version logic (#14725)

* Consolidate state version logic, add helpers to simplify updates

* Move version-related tests

* Use consistent casing

bitwiseguy committed:
contracts: use full compiler target for semver key (#14722)

* contracts: use full compiler target for semver key

* foundry: use_literal_content = true

* update check-semver-diff.sh filename parsing

* update semver-lock.json

* update check-frozen-files allow list to match compiler target names

* skip run-contracts-check ci job

* use negation to avoid deep nesting

bitwiseguy committed:
contracts: reenable semver-diff-check in ci (#14727)

mbaxter committed:
Refreeze MIPS contracts (#14732)

mbaxter committed:
cannon: Add more differential tests (#14684)

* Add a few more beq, bne test cases

* Tweak shift operation tests

* Add delay slot tests for jr/jalr

* Rework mov tests, add some test cases

* Add more slt(u) test cases

* Add slti tests

* Add sltiu tests

* Add test for sync instruction

* Rework binary operation tests

* Migrate nor test to use binary

* Fix comments

bitwiseguy committed:
op-deployer: add command to verify contracts (#14633)

* op-deployer: add 'verify' contracts command

* fix rate limiter values

* verified SuperchainConfigImplAddress

* verified superchain bundle (including constructor support)

* cleanup logs and flags

* cleanup: remove unused code

* verified implementations bundle

* verified opchain bundle

* update searchRemappings comment

* use more type-safe structs

* use l2-chain-id flag instead of l2-chain-index

* update to use new artifacts.Download function definition

* read constructor args from deployment tx initcode

* return error if contract creation query fails

* use more informative contract-name flag description

* calc num constructorArgSlots by parsing abi

* add json tags (matching contract bundle) to deploy script outputs

* bootstrap default to writing to bootstrap_<command>.json file

* pass contracts-locator flag to verify command

* minimize flags required for bootstrap, verify

* fix TestArtifactJSON

* add etherscan unit tests

* fix go lint errors

* make artifact lookup compatible with fragmented opcm

* fix go lint errors

* use require instead of assert in tests

mslipper committed:
op-validator: Add mainnet validators (#14734)

sebastianst committed:
all: Implement optional PectraBlobSchedule fork/feature (#14680)

* op-node/rollup: Implement optional L1 Pectra fix

* make simple test fail with excess blob gas

* improve test, fix L1 miner

* fix L1 miner again

* geoknee feedback

* update op-geth dependency to v1.101503.0-rc.2

* address proto's feedback, add test case

* fix static chain cfg tests

trianglesphere committed:
Don't split log.msg (#14735)

Splitting the message makes it hard to grep to find the source of the error.

ajsutton committed:
op-challenger: Interop vm runner (#14669)

* op-challenger: Add config option to set dependency set config

* Fix expected error message

* op-challenger: Update op-program executor to handle interop inputs.

* op-challenger: Support super-cannon in run trace.

* Fixes.

* Remove refactoring to introduce ToSuper()

mslipper committed:
op-deployer: Upgrade contracts support to v3.0.0 (#14733)

bitwiseguy committed:
op-deployer: add documentation for verify command (#14745)

* op-deployer: add documentation for verify command

* Update op-deployer/book/src/user-guide/verify.md

* Update verify.md

---------

Co-authored-by: Matthew Slipper <me@matthewslipper.com>

ajsutton committed:
ci: Fix preimage-reproducibility (#14742)

* ci: Capture reproducibility build logs as artifacts

* Install dependencies when mise.toml is present.

dependabot[bot] committed:
dependabot(gomod): bump github.com/prometheus/client_golang (#14628)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.20.5 to 1.21.1.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.20.5...v1.21.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pcw109550 committed:
kt-devnet: remove network params for supervisor (#14746)

pcw109550 committed:
op-service: Expand EthClient for tx handling (#14747)

protolambda committed:
op-supervisor,op-service: fix server-client API consistency (#14720)

* op-supervisor,op-service: fix server-client API consistency

* op-e2e: fix finality test, finalized-L1 getter now errors when not initialized

agusduha commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -137,12 +137,13 @@ contract Setup {
         // Just in case we have both FORK_TEST and L2_FORK_TEST set
-        return !isForkTest() && vm.envOr("L2_FORK_TEST", false);
+        return vm.envOr("FORK_TEST", false) && block.chainid != Chains.Sepolia && block.chainid != Chains.Mainnet
+            && block.chainid != Chains.Goerli;
```
hmm i think we should specifically tests the ones that are listed in the Chains lib, like `Chains.OPMainnet` or `Chains.OPGoerli`, if not then chains without predeploys will fail

i know we will leaving out the other L2s but once we get the standard genesis it will be the same

agusduha commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -161,7 +162,7 @@ contract Setup {
             );
-        } else if (isL2ForkTest()) {
+        } else if (isL2UpgradeForkTest()) {
             vm.createSelectFork(vm.envString("L2_FORK_RPC_URL"), vm.envUint("L2_FORK_BLOCK_NUMBER"));
```
The envs now are outdated here

agusduha commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -161,7 +162,7 @@ contract Setup {
                 block.chainid == Chains.Sepolia || block.chainid == Chains.Mainnet,
                 "Setup: ETH_RPC_URL must be set to a production (Sepolia or Mainnet) RPC URL"
```
is this require now?

agusduha commented on packages/contracts-bedrock/justfile:
```diff
@@ -84,6 +84,15 @@ prepare-upgrade-env *ARGS : build-go-ffi
+  export FORK_RPC_URL=$ETH_RPC_URL
+  export FORK_BLOCK_NUMBER=$pinnedBlockNumber
+  forge test {{ARGS}} \
+  --match-path "test/L2/forked/**"
```
I want to go a little further like the L1 forks tests, we should try integrating all `"test/L2/**"`

This way we are testing every L2 tests over the fork and ensure that everything works correctly

This is how L1 fork tests are working right now and we should try to emulate the same. There will be some failing tests because of predeploys not existing in mainnet, those ones should be skipped in its own test setup like the L1 tests do

agusduha commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -321,6 +332,16 @@ contract Setup {
         labelPreinstall(Preinstalls.HistoryStorage);
         labelPreinstall(Preinstalls.CreateX);
 
+        (bool success, bytes memory isIsthmusUpgradeActive) =
+            address(l1Block).staticcall(abi.encodeWithSelector(IL1Block.isIsthmus.selector));
+
+        // We need to be able to check whether the isthmus upgrade is active
+        // This being false means either the upgrade hasn't been activated or the chain was deployed after the upgrade
+        if (!success || isIsthmusUpgradeActive.length <= 0) {
```
Lets put this into an internal function that returns a flag

agusduha commented on packages/contracts-bedrock/test/setup/Setup.sol:
```diff
@@ -137,7 +137,13 @@ contract Setup {
     function isForkTest() public view returns (bool) {
-        return vm.envOr("FORK_TEST", false);
+        return vm.envOr("FORK_TEST", false) && (block.chainid == Chains.Sepolia || block.chainid == Chains.Mainnet);
+    }
```
Should this be called `isL1ForkTest` now and the new one `isL2ForkTest`?

ajsutton committed:
op-program: Update prestate reproducibility check to compare against list of prestates from superchain-registry (#14744)

* op-program: Switch prestate verification to use the superchain-registry as source of truth

* op-program: Update prestate reproducibility check to reference the same list superchain-registry now uses

* Fix script

* ci: Reduce timeout for prestate builds again.

* Clarify cannon 32 was built.

* Remove TODO

teddyknox committed:
Fix problem with kurtosis-devnet deploy (#14729)

0xOneTony committed:
fix: sync issues (#307)

* fix: merge issues

* fix: natspec

* fix: test

* fix: nit

zhiqiangxu committed:
allow specifying a custom package (#14750)

ajsutton committed:
op-program: Add a host subcommand to list available chain configs. (#14699)

* op-program: Add a host subcommand to list available chain configs.

* op-program: Also list custom chain configs

sebastianst committed:
go: update SR to cb1b645b35813b8dd3a900b9f99c46614e466a2c via op-geth (#14755)

* go: update SR to cb1b645b35813b8dd3a900b9f99c46614e466a2c via op-geth

to pull in updated sepolia-dev-0 sepolia blob schedule fix times

* fix chain config test

geoknee committed:
op-batcher: force use of TerminalString throughout syncActions logging (#14687)

Depending on the logger, it may not call this automatically

axelKingsley committed:
update TODO #11693 and #14765 (#14766)

Inphi committed:
op-program: Fix todo (#14773)

smartcontracts committed:
feat: interop portal updates (#14664)

* feat: AnchorStateRegistry as source of truth

Updates the OptimismPortal to use the AnchorStateRegistry as the
source of truth for the validity of Dispute Game contracts.

* feat: new eth lockbox (#285)

* feat: create new eth lockbox contract with interface

* chore: add it on the deployment scripts (wip)

* feat: create the test base to check if the setup for it is working

* feat: add all eth lockbox tests

* fix: opcm deployment script issue

* fix: lockbox tests

* feat: add no withdrawal tx unlock eth check

* chore: check proper initialization on test

* fix: pre-pr fixes

* chore: run pre-pr

* fix: opcm tests related to eth lockbox

* feat: add admin owner check on authorize portal and lockbox as well

* refactor: add prefix to errors and declare them on contract

* chore add no withdrawal tx comment

* feat: authorize portal in lockbox on the opcm

* chore: add one more check for eth lockbox

* fix: intializable and specs tests

* fix: pre-pr

* feat: integrate portal with lockbox (#291)

* feat: integrate portal with lockbox

* fix: migrate natspec

Co-authored-by: Disco <131301107+0xDiscotech@users.noreply.github.com>

* fix: delete natspec

Co-authored-by: Disco <131301107+0xDiscotech@users.noreply.github.com>

* fix: natspec nits

* chore: natspec wording

* chore: function order

* chore: remove internals

* feat: add eth lockbox as portal arg on opcm

* feat: add checks on scripts and unit tests

* feat: lock and unlock tests

* feat: add migrate liquidity tests

* fix: other failing portal tests

* chore: run pre-pr

* chore: improve lock and unlock checks on tests

* chore: undo portal version change and run pre-pr

* refactor: enhance expect call checks

* fix: fork test string error

* fix: just test failing tests

* refactor: pao base

* refactor: setup portal and lockbox integration on initialization

* feat: update lockbox setter

* chore: update pao naming on tests

* chore: portal comment

* chore: update version and run pre-pr

* fix: spec

* fix: lockbox storage layout

* fix: pr comments and failing test

* feat: portal upgrade test

* chore: remove portal balance check on chain assertions

* fix: error string on chain assertions

* feat: add test for portal unsafe target

* fix: approval

---------

Co-authored-by: Disco <131301107+0xDiscotech@users.noreply.github.com>

* fix: semver lock

* fix: failing tests after merge

* fix: interface import

* chore: add lockbox pao matches final system owner check

* fix: pr comments (#293)

* chore: remove unused function

* chore: add pao validation checks over portal and lockbox

* refactor: use interface as arg type instead of address on functions

* fix: comment max length

* refactor: update lockbox param type as interface

* refactor: use interface as type on migrate liquidity and authorize lockbox functions

* chore: run pre pr

* fix: high fuzz runs failing test using mostly assume not forge address

* refactor: use ioptimis portal instead of ioptimism portal2 everywhere

* chore: undo changes on unrelated files to the pr

* feat: add Super Root specific method to OptimismPortal

Updates the OptimismPortal to include a method that allows users
to prove against Super Roots for interop.

* fix: pr second comments (#295)

* chore: undo initializable v5 test changes

* chore: remove portal already authorized check

* chore: remove portal already authorized check

* refactor: rename pao to proxy admin owner

* fix: pre pr

* fix: test upgrade fails (#296)

* fix: test upgrade fails

* feat: handle fork state on eth lockbox tests

* chore: enhance comments

* chore: run pre-pr

* fix: semgrep

* fix: comments

* chore: address path where is not a fork to get the lockbox

* chore: run pre pr

* fix: pr fixes (#298)

* chore: enhance opcm comment

* fix: same proxy admin owner typo

* feat: add insufficient balance check

* chore: add natspec comments on lockbox migration process

* chore: pre pr

* fix: add lockbox on implementations struct to fix go test (#300)

* feat: AnchorStateRegistry as source of truth

Updates the OptimismPortal to use the AnchorStateRegistry as the
source of truth for the validity of Dispute Game contracts.

* feat: add Super Root specific method to OptimismPortal

Updates the OptimismPortal to include a method that allows users
to prove against Super Roots for interop.

* feat: interop portal OPCM updates

* fix: remove old respectedGameType check from portal test

* fix: deploy new dispute games in OPCM

* feat: add upgrade 15 test path

* feat: integrate into opcm fork tests

* fix: corrected checks for ASR

* fix: rebase tweaks

* fix: OPCM integration

* fix: OPCM stack too deep

* fix: pre pr

* fix: update portal implementation on opcm upgrade

* fix: call upgrade on portal without upgrading any impl

* fix: iopcm interface for older versions already deployed

* refactor: use minimal interface for opcm without lockbox

* fix: import

* chore: undo changes on opcm carried when resolving conflicts (#302)

* fix: include lockbox on reinitialization test and add todos with issue number (#303)

* fix: include lockbox on reinitialization test

* fix: polish nits

* refactor: get eth lockbox on test condition

* chore: add todos pointing to the issue

* fix: add ETHLockbox to op-deployer

* fix: add evm tags to ETHLockboxImpl

* fix: evm tags for ethLockboxProxy too

* fix: remove certain ETHLockbox checks

* fix: properly add ETHLockbox to state

* fix: handle deposit tests and kill L2oo tests

* fix: undo change in user test

* fix: kill more L2OO tests

* fix: bug in withdrawal validity test

* fix: pr review (#309)

* chore: remove unnecessary cast

* refactor: add lockbox on eth migrated event

* chore: rename pa owner to pa owned base

* refactor: add amount on liquidity received and migrated events

* fix: format

* refactor: add amount on liquidity migrated event

* chore: pre-pr

---------

Co-authored-by: agusduha <agusnduha@gmail.com>

* fix: some remaining merge issues

* fix: go linting error

* fix: remove old test skips

* fix: unskip v2 upgrade tests

* fix: skip v2 upgrade tests again

---------

Co-authored-by: Disco <131301107+0xDiscotech@users.noreply.github.com>
Co-authored-by: AgusDuha <81362284+agusduha@users.noreply.github.com>
Co-authored-by: agusduha <agusnduha@gmail.com>

0xOneTony committed:
fix: tests

axelKingsley committed:
Add consistency checks in ChainsDB Queries (#14769)

scharissis committed:
feat(op-acceptance-tests): introduce op-acceptance-tests. (#14706)

This introduces a new directory for acceptance tests. Within it we can store our tests, the op-acceptor configuration and a justfile to simplify running the acceptance tests.

ajsutton committed:
op-program: Update issue number on TODO (#14776)

0xOneTony committed:
Merge pull request #306 from defi-wonderland/chore/sync

chore: sync develop

geoknee committed:
op-batcher: introduce `ClearAllStateMetrics()` and call from `channelManager.Clear()` (#14780)

* op-batcher: introduce ClearAllStateMetrics() and call from channelManager.Clear()

* fix test metrics

* use real metrics in test

* add godoc

0xOneTony committed:
Merge branch 'develop' of https://github.com/defi-wonderland/optimism into chore/sync-lockbox

0xOneTony committed:
fix: pre-pr and versions

0xOneTony committed:
fix: test

0xOneTony committed:
chore: sync lockbox changes

chore: sync lockbox

0xiamflux committed:
test: update naming and conditions for L1 & L2 fork tests

0xiamflux committed:
test: move upgrade check to its own internal function

0xiamflux committed:
test: update tests

0xiamflux committed:
test: expand l2 fork tests scope

0xiamflux committed:
Merge branch 'sc-feat/standard-l2-genesis' into test/l2-fork-tests

agusduha commented:
Closing until L2 NUTs testing project is resumed, tracking in it's [internal page](https://www.notion.so/defi-wonderland/L2-NUTs-Testing-Internal-2229a4c092c78092b53bcaf6aec14585?source=copy_link)
